presto-main/src/main/java/com/facebook/presto/operator/exchange/LocalExchange.java

```diff
@@ -121,7 +129,13 @@ else if (partitioning.equals(FIXED_PASSTHROUGH_DISTRIBUTION)) {
             };
         }
         else {
-            throw new IllegalArgumentException("Unsupported local exchange partitioning " + partitioning);
+            // partitioned exchange
+            exchangerSupplier = () -> new PartitioningExchanger(
```

Comment:
Maybe assert `partitionFunction` doesn't present for other type of `partitioning` ?

Reply:
Computing it here now